python/cugraph/graph/convert_matrix.pyx

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this file is pure python and no need to be a cython file. Once cugraph's
+# issue #146 is addressed, this file's extension should be changed from .pyx to
+# .py and should be located outside the python/cugraph/bindings directory.
+
+import cudf
+
+from cugraph import Graph
+
+
+def from_cudf_edgelist(df, source='source', target='target', weight=None):
+    """
+    Add edges to a graph. Currently, self should be an empty graph.
```

Comment:
I'm confused.  The comment suggests that an error will be raised if self is not empty.  The code does nothing with self, creates an empty graph, adds edges and returns the new graph object.  This seems inconsistent.

Reply:
A great catch, I forgot to update comments after moving this from inside the Graph class (as a member function) to outside the Graph class. I will update the comments.

Reply:
And I should remove add_edges_from() and add_weighted_edges_from() as discussed in the email, so no need to review those functions (I didn't know this PR will be reviewed at this point, I was waiting for cudf's fix so all the tests can be passed before starting a review).